src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -11448,19 +11531,58 @@ def update_tag(
 
         Returns:
             An updated tag.
+
+        Raises:
+            RuntimeError: If the tag can not be converted to an exclusive tag due
+                to it being associated to multiple entities.
         """
         with Session(self.engine) as session:
             tag = self._get_tag_schema(
                 tag_name_or_id=tag_name_or_id,
                 session=session,
             )
-
             self._verify_name_uniqueness(
                 resource=tag_update_model,
                 schema=tag,
                 session=session,
             )
 
+            if tag_update_model.exclusive is True:
+                error_messages = []
+                for resource_type, resource_id, scope_id in [
+                    (
+                        TaggableResourceTypes.PIPELINE_RUN,
+                        PipelineRunSchema.id,
+                        PipelineRunSchema.pipeline_id,
+                    ),
+                    (
+                        TaggableResourceTypes.ARTIFACT_VERSION,
+                        ArtifactVersionSchema.id,
+                        ArtifactVersionSchema.artifact_id,
+                    ),
+                    (
+                        TaggableResourceTypes.RUN_TEMPLATE,
+                        RunTemplateSchema.id,
+                        RunTemplateSchema.id,
```

Comment:
```suggestion
                        RunTemplateSchema.pipeline_id,
```

I assume?

Reply:
Correct. However, this was a bit trickier to solve. `RunTemplateSchema` does not have a `pipeline_id`. That means I had to use the deployment in between. Accordingly, I had to adjust the duplicate check that we use when we are updating tags.